### PR TITLE
Kai UI Federated Store Request Fix + other refactoring

### PR DESCRIPTION
**Pull request by K-itKat20**
This pull request includes:

- #249 Refactoring Kai UI to send create graph request for federated graphs correctly
- The error handling + notifications in Kai UI has also been refactored.
- The GaaS rest has also been updated to send the Rest and UI URLs for graphs, to be displayed in the view graphs section of Kai UI
- The unit tests in Kai UI have been refactored to accurately reflect the types of Graph UI and Rest URLs  that it will receive

### CODE REVIEW

File: gaas-ui/test/components/create-graph/create-graph.test.tsx [hunk 1]
```diff
@@ -175,7 +175,7 @@ describe("CreateGraph UI component", () => {
             mockGetGraphDescription("Description for this Proxy Graph");
             mockGetGraphId("graph-id");
 
-            await inputProxyURL("http://test.graph.url");
+            await inputProxyURL("http://http://resoucename-namespace.host-name/rest");
```
Comment (t92549) [inline gaas-ui/test/components/create-graph/create-graph.test.tsx:178]: Why does it say http twice?

**Review by t92549 — COMMENTED** [review]
Thanks for the PR!

File: gaas-ui/test/components/create-graph/create-graph.test.tsx (hunk 1 above)
Comment (K-itKat20) [reply gaas-ui/test/components/create-graph/create-graph.test.tsx]: fixed

File: gaas-ui/server/middleware.js [hunk 2]
```diff
@@ -70,65 +70,74 @@ app.get("/graphs", (req, res) => {
...
                         description: "Secondary development mode graph",
-                        url: "http://dev-environment-2.k8s.cluster/rest",
+                        url: "http://roadtraffic6-namespace.host-name/ui",
+                        restUrl: "http://roadtraffic6-namespace.host-name/rest",
                         configName: "mapStore",
                         status: "UP",
                     },
                     {
-                        graphId: "testGaffer",
+                        graphId: "roadtraffic6",
```
[the remark this answers is not in the document]
Comment (K-itKat20) [reply gaas-ui/server/middleware.js]: refactored

**Review by t92549 — APPROVED** [review]

**Comment by t92549** [comment]
@K-itKat20 ready to merge when you are

**Comment by rocky341** [comment]
@t92549 
Ready to merge please

**Pull request merged**